lms/static/js/student_account/views/RegisterView.js

```diff
@@ -133,32 +282,81 @@
                 getFormData: function() {
                     var obj = FormView.prototype.getFormData.apply(this, arguments),
                         $form = this.$form,
-                        $label,
-                        $emailElement,
-                        $confirmEmailElement,
-                        email = '',
-                        confirmEmail = '';
-
-                    $emailElement = $form.find('input[name=email]');
-                    $confirmEmailElement = $form.find('input[name=confirm_email]');
-
-                    if ($confirmEmailElement.length) {
-                        email = $emailElement.val();
-                        confirmEmail = $confirmEmailElement.val();
-                        $label = $form.find('label[for=' + $confirmEmailElement.attr('id') + ']');
-
-                        if (confirmEmail !== '' && email !== confirmEmail) {
-                            this.errors.push('<li>' + $confirmEmailElement.data('errormsg-required') + '</li>');
-                            $confirmEmailElement.addClass('error');
-                            $label.addClass('error');
-                        } else if (confirmEmail !== '') {
-                            obj.confirm_email = confirmEmail;
-                            $confirmEmailElement.removeClass('error');
-                            $label.removeClass('error');
+                        $emailElement = $form.find('input[name=email]'),
+                        $confirmEmail = $form.find('input[name=confirm_email]'),
+                        elements = $form[0].elements,
+                        $el,
+                        key = '',
+                        i;
+
+                    for (i = 0; i < elements.length; i++) {
+                        $el = $(elements[i]);
+                        key = $el.attr('name') || false;
+
+                        // Due to a bug in firefox, whitespaces in email type field are not removed.
+                        // TODO: Remove this code once firefox bug is resolved.
+                        if (key === 'email') {
+                            $el.val($el.val().trim());
+                        }
+
+                        // Simulate live validation.
+                        if (this.inValidationFields($el)) {
+                            $el.blur();
+
+                            // Special case: show required string for errors even if we're not focused.
+                            if ($el.hasClass('error')) {
+                                this.renderRequiredMessage($el);
+                            }
                         }
                     }
 
+                    if ($confirmEmail.length) {
+                        if (!$confirmEmail.val() || $emailElement.val() !== $confirmEmail.val()) {
```

Comment:
So I've been following a pattern (as it is in other places in the codebase and I believe with JS devs who use jquery in general) where anything prefixed with a `$` is a reference to an HTML element. So when I check for the length, I'm really seeing if the object that represents this element exists at all. Then I check for the value, which is an individual key-value pair in that object. This style is used in other places and yes, for JS in general I believe, to check for existence.

As for the order of operations, sure, I can put that last part into parantheses.